Site/SiteMediaDurationUpdater.php

```diff
@@ -0,0 +1,277 @@
+	{
+		parent::__construct($id, $config_filename, $title, $documentation);
+
+		$this->addCommandLineArgument(
+			new SiteCommandLineArgument(
+				array('--dry-run'),
+				'setDryRun',
+				Site::_(
+					'Durations are only calculated but not saved. Use '.
+					'with --verbose to see what durations differ.'
+				)
+			)
+		);
+
+		$shortname = new SiteCommandLineArgument(
+			array('-s', '--shortname'),
+			'setMediaSetShortname',
+			Site::_(
+				'Sets the shortname of the media set whose durations we want '.
+				'to recalculate.'
+			)
+		);
+
+		$shortname->addParameter(
+			'string',
+			'shortname name must be specified.'
+		);
+
+		$this->addCommandLineArgument($shortname);
+	}
+
+	// }}}
+	// {{{ public function run()
+
+	public function run()
+	{
+		$this->initModules();
+		$this->parseCommandLineArguments();
+
+		$this->lock();
+
+		$now = new SwatDate();
+		$now->toUTC();
+
+		foreach ($this->getMedia() as $media) {
+			$this->updateDuration($media);
+		}
+
+		$this->unlock();
+		$this->debug("done\n");
+	}
+
+	// }}}
+	// {{{ public function setDryRun()
+
+	public function setDryRun($dry_run)
+	{
+		$this->dry_run = (boolean)$dry_run;
+	}
+
+	// }}}
+	// {{{ public function setMediaSetShortname()
+
+	public function setMediaSetShortname($shortname)
+	{
+		$this->media_set_shortname = $shortname;
+	}
+
+	// }}}
+	// {{{ public function setMediaFileBase()
+
+	public function setMediaFileBase($media_file_base)
+	{
+		$this->media_file_base = $media_file_base;
+	}
+
+	// }}}
+	// {{{ protected function getMediaSet()
+
+	protected function getMediaSet()
+	{
+		if ($this->media_set_shortname == '') {
+			throw new SiteException('A media set shortname must be specified.');
+		}
+
+		$class_name = SwatDBClassMap::get('SiteMediaSet');
+
+		$media_set = new $class_name();
+		$media_set->setDatabase($this->db);
+
+		if (!$media_set->loadByShortname($this->media_set_shortname)) {
+			throw new SiteException(
+				sprintf(
+					'Unable to load media set with shortname “%s”.',
+					$this->media_set_shortname
+				)
+			);
+		}
+
+		return $media_set;
+	}
+
+	// }}}
+	// {{{ protected function getMedia()
+
+	protected function getMedia()
+	{
+		$sql = sprintf(
+			'select * from Media where media_set = %s order by id',
+			$this->db->quote($this->getMediaSet()->id, 'integer')
+		);
+
+		$audio = SwatDB::query(
+			$this->db,
+			$sql,
+			SwatDBClassMap::get('SiteMediaWrapper')
+		);
+
+		return $audio;
+	}
+
+	// }}}
+	// {{{ protected function updateDuration()
+
+	protected function updateDuration(SiteMedia $media)
+	{
+		if ($this->media_file_base == '') {
+			throw new SiteException('A media file base must be specified.');
+		}
+
+		$media->setFileBase($this->media_file_base);
+
+		$filename = $media->getFilePath('original');
+
+		if (file_exists($filename)) {
+			$old_duration = $media->duration;
+			$new_duration = SiteAudioMedia::parseDuration($filename);
+
+			$this->debug(
+				sprintf(
+					"Updating Media %s...",
```

Comment:
For other command-line apps we've translated debug output with `Site::_()`. See `SiteCdnUpdater`